src/storage/src/hummock/file_cache/recent_filter.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{BTreeSet, VecDeque};
+use std::fmt::Debug;
+use std::time::{Duration, Instant};
+
+use parking_lot::RwLock;
+
+#[derive(Debug)]
+pub struct RecentFilter<K>
+where
+    K: Eq + Ord + Send + Sync + Debug + Clone + 'static,
```

Comment:
may introduce `trait RecentFilterKey = Eq + Ord + Send + Sync + Debug + Clone + 'static`, and then here only use `K: RecentFilterKey`.